code/Style.md

```diff
@@ -0,0 +1,227 @@
+
+The following is roughly in the order of priority.
+
+If the guide is not followed then CI may fail.
+
+## Formatting
+
+All code is formatted with `goimports -local storj.io`. Where `goimports` is using the latest supported stable Go version.
+
+### Imports
+
+Import statements are in three groups in the following order:
+
+```
+import (
+	// standard packages
+	"context"
+	...
+	"sort"
+
+	// external packages
+	"github.com/gogo/protobuf/proto"
+	...
+	"go.uber.org/zap"
+
+	// storj.io packages
+	"storj.io/storj/internal/fpath"
+	...
+)
+```
+
+## Linting
+
+All code should pass the linters setup in the project folder.
+
+## Copyright
+
+All code must have a copyright header.
+
+## Error handling
+
+All errors must be handled and checked. Don't hide errors.
+
+_Exceptions: printing to console or log (e.g. `fmt.Println`, `log.Print`)_
+
+Use `errs.Class` to annotate errors with more information.
+
+To combine multiple errors use `errs.Combine`. To collect multiple similar errors use `errs.Group`.
+
+To handle errors with things that need closing use `errs.Combine` together with `defer`. As an example:
+
+```
+var Error = errs.Class("example error")
+
+func Example() (err error) {
+	file, err := os.Open("example.txt")
+	if err != nil {
+		return Error.Wrap(err)
+	}
+	defer func() {
+		err = errs.Combine(err, Error.Wrap(file.Close()))
+	}
+
+	return Process(file)
+}
+```
+
+## Comments
+
+See [Comment Sentences](https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences). Comments should help the reader to orient to the constraints and relations with other packages.
+
+Bad example:
+
+```
+// Encode encodes req to w.
+func Encode(w io.Writer, req *Request)
+```
+
+Better example:
+
+```
+// Encode encodes req as JSON to w, which is used by Server.
+func Encode(w io.Writer, req *Request)
+```
+
+Avoid tautological comments:
+
+```
+// DB is a database.
+type DB interface {
+```
+
+Better example:
+
+```
+// DB stores raw accounting information gathered by tally.
+type DB interface {
+```
+
+Comments should have a space after `//`, e.g. `// DB` instead of `//DB`.
+
+## Logging
+
+Use properly namespaced `*zap.Logger`. By properly namespacing and passing in the logger we can better find how things are working. Tests should use `zaptest.NewLogger(t)` as the root logger.
+
+Avoid using global loggers (e.g. `zap.L().Error`, `log.Print`, `fmt.Print`), unless it is `package main`.
+
+## Variable naming
+
+Avoid single letter receivers. Prefer hard-to-confuse variables with 3-7 letters. Use conventional naming when there is one (e.g. `sync.Mutex` is usually called `mu`.) For variables a single letter variable can be fine only if the scope doesn't exceed one page (about 50 lines).
+
+Single letter receivers can be convenient, however when moving between a lot of packages it can be quickly confusing.
```

Comment:
this isn't idiomatic advice: https://talks.golang.org/2014/names.slide#11. i'm okay with ignoring idiomatic advice but then i think i have a stronger case for 80 char column width limits :)

Reply:
aslo https://research.swtch.com/names

Reply:
Yeah, I'm aware, however given the current project I think erring on the side of slightly longer names doesn't cause problems.